Enable resource builder flattening in rollout status

ardaguclu opened the pull request:
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
To make selector flag is working in `rollout status` command, we need to flatten resources in builder like it is done in other `rollout` commands (ie. (https://github.com/kubernetes/kubernetes/blob/c4f8c57b43843abd13bae677b73c622e1c30b46f/staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_history.go#L150)

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubectl/issues/1223

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

ardaguclu committed:
Enable resource builder flattening in rollout status

eddiezane reviewed (approved):
Thank you @ardaguclu! May I please ask you to add tests for rollout status with this PR?

ardaguclu committed:
Add unit tests for rollout status command

This commit makes adjustments in rollout status command to make it
unit testable. In addition to that, unit tests for rollout status
command is added.

ardaguclu committed:
Add license header into rollout status test file

ardaguclu commented:
/test pull-kubernetes-integration

pacoxu commented on staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go:
```diff
@@ -170,6 +165,7 @@ func (o *RolloutStatusOptions) Run() error {
 		ResourceTypeOrNameArgs(true, o.BuilderArgs...).
 		ContinueOnError().
 		Latest().
+		Flatten().
```
LGTM

pacoxu reviewed (approved):
Most scenarios are coved by test cases.

ardaguclu commented:
@eddiezane @pacoxu would you mind tagging this PR?. Thanks.

pacoxu commented:
/lgtm
